pkg/controller/jobframework/reconciler.go

```diff
@@ -731,29 +751,36 @@ func FindMatchingWorkloads(ctx context.Context, c client.Client, job GenericJob)
 	return match, toDelete, nil
 }
 
-func (r *JobReconciler) ensurePrebuiltWorkloadOwnership(ctx context.Context, wl *kueue.Workload, object client.Object) (bool, error) {
+func EnsurePrebuiltWorkloadOwnership(ctx context.Context, c client.Client, wl *kueue.Workload, object client.Object) (bool, error) {
 	if !metav1.IsControlledBy(wl, object) {
-		if err := ctrl.SetControllerReference(object, wl, r.client.Scheme()); err != nil {
-			// don't return an error here, since a retry cannot give a different result,
-			// log the error.
-			log := ctrl.LoggerFrom(ctx)
-			log.Error(err, "Cannot take ownership of the workload")
-			return false, nil
```

Comment:
Actually, I like the fact we don't silence the error, but it seems a little bit disconnected from the PR. Could you send a separatory / preparatory PR which changes that? I think it might actually be a bugfix. Can you think of a user-facing scenario? If so, I believe we could even consider a cherry-pick.

Reply:
Done on https://github.com/kubernetes-sigs/kueue/pull/4069.